check-submodules.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Checks out if a submodule has been updated
+SUBMODULES=`git submodule--helper list | awk '{ print $4 }'`
+RUN_CI="no"
+for SUBMODULE in $SUBMODULES
+do
+    git diff --exit-code HEAD^ $SUBMODULE
+    CHANGED=$? # Get git diff exit code
+    if [ $CHANGED -eq 1 ]
+    then
+        RUN_CI="yes"
+        SUBMODULE_NAME=$submodule
+        break
+    fi
+done
+
+# If no submodule has been updated, exit the script
+if [ "$RUN_CI" = "no" ]; then
+    exit 0
+fi
+
+# Install json-diff
+JSON_DIFF_LINK="https://github.com/Luni-4/json-diff/releases/download"
+JSON_DIFF_VERSION="0.0.1"
+curl -L "$JSON_DIFF_LINK/v$JSON_DIFF_VERSION/json-diff-linux.tar.gz" |
+tar xz -C $CARGO_HOME/bin
+
+# Install json minimal tests
+JMT_LINK="https://github.com/Luni-4/json-minimal-tests/releases/download"
+JMT_VERSION="0.0.1"
+curl -L "$JMT_LINK/v$JMT_VERSION/json-minimal-tests-linux.tar.gz" |
+tar xz -C $CARGO_HOME/bin
+
+# Download mozilla-central repository
+MOZILLA_CENTRAL_REPO="https://github.com/mozilla/gecko-dev"
+git clone --depth 1 -j8 $MOZILLA_CENTRAL_REPO $HOME/mozilla-central
```

Comment:
You can reuse the commands we use in the `test_mozilla_central` task to avoid recloning if the repo is already there